src/sentry/notifications/utils/__init__.py

```diff
@@ -68,9 +68,7 @@ def get_release(activity: Activity, organization: Organization) -> Release | Non
         return None
 
 
-def get_group_counts_by_project(
-    release: Release, projects: Iterable[Project]
-) -> Mapping[Project, int]:
+def get_group_counts_by_project(release: Release, projects: Iterable[Project]) -> dict[int, int]:
```

Comment:
sort of unintuitively -- in django's orm `.values_list("foreign_key_field")` produces a sequence of the key type and not the value type -- fixing this fixed some usages of this dictionary in the PR targeted file